mcs/class/Mono.Profiler.Log/Mono.Profiler.Log/LogEventPool.cs

```diff
@@ -0,0 +1,36 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Mono.Profiler.Log {
+
+	sealed class LogEventPool {
```

Comment:
What about changing the logic to use more delay-loading with something like

```csharp
struct MethodCoverageLogEvent
{
    internal long streamPosition;
    MethodCoverageLog ReadData (DataManager dataManager)
    {
        // initialize from stream in dataManager
    }
}

public class MethodCoverageLog
{
    // Record class for method coverage data
}
```

You keep cyclic array of WeakMethodCoverageLogEvent which you reuse after